pkg/trace/stats/oteltest/testutil.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2016-present Datadog, Inc.
+
+package oteltest
+
+import (
+	"context"
+	"net/http"
+	"runtime"
+	"sync"
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/agent"
+	"github.com/DataDog/datadog-agent/pkg/trace/api"
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-agent/pkg/trace/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/trace/timing"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// This is a copy of https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/datadog/agent.go and
+// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/datadog/metrics_client.go.
+// This fake agent is only used in tests.
+
+// TraceAgent specifies a minimal trace agent instance that is able to process traces and output stats.
+type TraceAgent struct {
+	*agent.Agent
+
+	// pchan specifies the channel that will be used to output Datadog Trace Agent API Payloads
+	// resulting from ingested OpenTelemetry spans.
+	pchan chan *api.Payload
+
+	// wg waits for all goroutines to exit.
+	wg sync.WaitGroup
+
+	// exit signals the agent to shut down.
+	exit chan struct{}
+}
+
+// newAgentWithConfig creates a new traceagent with the given config cfg. Used in tests; use newAgent instead.
+func NewAgentWithConfig(ctx context.Context, cfg *traceconfig.AgentConfig, out chan *pb.StatsPayload, now time.Time) *TraceAgent {
+	// disable the HTTP receiver
+	cfg.ReceiverPort = 0
+	// set the API key to succeed startup; it is never used nor needed
+	cfg.Endpoints[0].APIKey = "skip_check"
+	// set the default hostname to the translator's placeholder; in the case where no hostname
+	// can be deduced from incoming traces, we don't know the default hostname (because it is set
+	// in the exporter). In order to avoid duplicating the hostname setting in the processor and
+	// exporter, we use a placeholder and fill it in later (in the Datadog Exporter or Agent OTLP
+	// Ingest). This gives a better user experience.
+	cfg.Hostname = "__unset__"
+	pchan := make(chan *api.Payload, 1000)
+	metricsClient := &statsd.NoOpClient{}
+	a := agent.NewAgent(ctx, cfg, telemetry.NewNoopCollector(), metricsClient)
+	// replace the Concentrator (the component which computes and flushes APM Stats from incoming
+	// traces) with our own, which uses the 'out' channel.
+	a.Concentrator = stats.NewConcentrator(cfg, out, now, metricsClient)
+	// ...and the same for the ClientStatsAggregator; we don't use it here, but it is also a source
+	// of stats which should be available to us.
+	a.ClientStatsAggregator = stats.NewClientStatsAggregator(cfg, out, metricsClient)
+	// lastly, start the OTLP receiver, which will be used to introduce ResourceSpans into the traceagent,
```

Comment:
this comment says start the OTLP receiver, but it appears we just initialize one here and don't actually call `Start`

Reply:
It's a copy-paste, fixed.